flink-kubernetes-operator/src/test/java/org/apache/flink/kubernetes/operator/reconciler/deployment/ApplicationReconcilerTest.java

```diff
@@ -1223,7 +1224,7 @@ public void testRestartUnhealthyEvent() throws Exception {
         var clusterHealthInfo = new ClusterHealthInfo();
         clusterHealthInfo.setTimeStamp(System.currentTimeMillis());
         clusterHealthInfo.setNumRestarts(2);
-        clusterHealthInfo.setHealthy(false);
+        clusterHealthInfo.setHealthResult(ClusterHealthResult.error("error"));
```

Comment:
We should extend the assertion in this method to make sure that the triggered event has the correct error message